### ISSUE TITLE
PyQt4: QString encoding problems - Win8.1, Python3.4

### ISSUE THREAD

**Issue by janpipek** [ContinuumIO/anaconda-issues#419]
I am using QtGui.QTextEdit to edit text, however I observed a problem when non-ASCII characters  (usually "central european" ones like "áčďéěíňóřšťúůýž") occured in it. I tried to test it down to QString level (using sip API 1) and it occures there as well.

Depending on a few conditions, the string returned from QString is either ok or misinterpreted. It seems to me as if the process first tried to convert the content first to some guessed encoding and then back to unicode. If the first non-ASCII character in the text is in the central-european set but not in the Western one (like "ř, š, ..."), the resulting string is ok. However, if the first non-ASCII is in Western charset, the following central european ones are misinterpreted There must be some strange encoding heuristics behind the scene.

I am incuding an example script that demonstrates this problem:

```python
# -*- coding: utf-8  -*-
import sys
import sip
sip.setapi('QString', 1)       # set it to 1 or 2
from PyQt4 import QtCore, QtGui

app = QtGui.QApplication(sys.argv)
edit = QtGui.QTextEdit()

# Two different cases:
for t in [u"ář", u"řá"]:
    # Write text to editwidget and retrieve it back
    edit.setPlainText(t)
    print("QTextEdit: expected %s, obtained %s" % (t, edit.toPlainText()))

    # Just convert a string there and back
    if sip.getapi('QString') == 1:
        converted = QtCore.QString(t)
        print("QString: expected %s, obtained %s" % (t, converted))
```

Output (in my case):

```
QTextEdit: expected ář, obtained áY
QString: expected ář, obtained áY
QTextEdit: expected řá, obtained řá
QString: expected řá, obtained řá
```

Maybe it's just a problem in PyQt4 itself, but I still think it is worth mentioning.

**Comment by ccordoba12** [comment]
Why don't you use API 2 instead of 1?

**Comment by janpipek** [comment]
Hi, the occurrence of problem does not depend on the API version (as I believe underneath the same C++ function gets called). In fact, I started naturally with API 2, but then only API 1 enabled me to create QString objects (to rule out it being a problem of QTextEdit).

However, I can now add, that in Linux, both 2.7 and 3.4 Pythons behave correctly. I'll try to check win 8.1/2.7 soon.

**Comment by janpipek** [comment]
In Windows 8.1, Python 2:7, there is no problem.

**Comment by ccordoba12** [comment]
Maybe an update to PyQt4 will fix this problem. I'm planning to do one next month.

**Comment by janpipek** [comment]
Seems solved with PyQt4 that is part of Anaconda 2.4.1.

**Comment by ccordoba12** [comment]
Ok
